### ISSUE TITLE
Support for scoped stubbing with the JUnit5 Extension.

### ISSUE THREAD

**Issue by rob-pomelo** [marcinziolo/kotlin-wiremock#14]
Instead of extending WireMockServer, the Junit5 Extension extends the [DslWrapper](https://sourcegraph.com/github.com/wiremock/wiremock/-/blob/src/main/java/com/github/tomakehurst/wiremock/junit/DslWrapper.java?L38:14). It would be great to support this in the DSL as well (instead of needing to use the static DSL).

Thanks for a great library!

**Comment by checketts** [comment]
I've tried to address this by #17 

I can't find any usage of `DslWrapper` on the '[using JUnit 5](https://wiremock.org/docs/junit-jupiter/)' page.

**Comment by checketts** [comment]
I'm asking for a link on how it is *used*. Digging into it, that appears to be for JUnit 4.

**Comment by rob-pomelo** [comment]
Ah gotcha, on the "using JUnit 5" page you linked in the "programmatic" section that is all enabled by the fact the extension extends DslWrapper (for example the `stubFor` method). The issue right now is that [WireMockInstance](https://github.com/marcinziolo/kotlin-wiremock/blob/master/src/main/kotlin/com/marcinziolo/kotlin/wiremock/WireMockInstance.kt#L21)  can be built from WireMockServer (which the JUnit4 rule extends) but not WiremockExtension (the Junit5 extension, because it extends DslWrapper instead). With an implementation of WireMockInstance for DslWrapper, e.g. something like:

```
class WiremockServerInstance(private val dslWrapper: DslWrapper): WireMockInstance() {
    override fun stubFor(mappingBuilder: MappingBuilder) = dslWrapper.stubFor(mappingBuilder)
    override fun removeStubMapping(stubMapping: StubMapping) = dslWrapper.removeStub(stubMapping)
    override fun getSingleStubMapping(uuid: UUID) = dslWrapper.getSingleStubMapping(uuid)
}
```

I think it would then be possible to use scoped mocks (e.g. `wiremock.post {`) with the Junit5 extension.

**Comment by rob-pomelo** [comment]
Update: Looking at your PR that would likely also be something we could do for our use as well, e.g.:
```
@RegisterExtension
  static WireMockExtension wm = WireMockExtension.newInstance().proxyMode(true)
  
  fun test() {
    wm.runtimeInfo().wireMock.get {..}
  }
  ```
  which works, just a bit more verbose than:
  ```
    fun test() {
    wm.get {..}
  }
  ```

**Comment by checketts** [comment]
So currently `wm` is a parameter that the `@WiremockTest` Junit 5 extension provides:

```kotlin
@WireMockTest
class ExampleExtensionTest {

    lateinit var wm: String

    @BeforeEach
    fun urlSetup(wmRuntimeInfo: WireMockRuntimeInfo) {
        wm = wmRuntimeInfo.wireMock             <----------------- Getting the wiremock instance
    }

    @Test
    fun `url equalTo`() {
        wm.get {     //     <----- Can use Wiremock instance directly
            url equalTo "/users/1"
        } returns {
            header = "Content-Type" to "application/json"
            statusCode = 200
            body = """
            {
              "id": 1,
              "name": "Bob"
            }
            """
        }

        When {
            get("$url/users/1")
        } Then {
            statusCode(200)
            body("id", equalTo(1))
            body("name", equalTo("Bob"))
        }
    }
}
```

Can you help me understand how scoped mocks works? I'm trying to write a test that uses it, but can't figure it out.

**Comment by marcinziolo** [comment]
Thanks @checketts  @rob-pomelo , Issue is fixed in 1.0.4
